.github/workflows/main.yml

```diff
@@ -387,14 +387,70 @@ jobs:
       - name: Build nbms
         run: ant $OPTS build-nbms
 
-      - name: Build source zips
-        run: ant $OPTS build-source-zips
-
       - name: Build javadoc
         if: env.test_javadoc == 'true' && success()
         run: ant $OPTS build-javadoc
 
 
+  build-from-src-zip:
+    name: Build ${{ matrix.config }} from src.zip on JDK ${{ matrix.java }}
+    needs: base-build
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    strategy:
+      matrix:
+        java: [ '11' ]
+        config: [ 'platform', 'release' ]
+    steps:
+
+      - name: Set up JDK ${{ matrix.java }} 
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ matrix.java }} 
+          distribution: ${{ env.default_java_distribution }}
+
+      - name: Download Build
+        uses: actions/download-artifact@v3
+        with:
+          name: build
+
+      - name: Extract
+        run: tar --zstd -xf build.tar.zst
+
+      - name: Restoring Cache
+        uses: actions/cache/restore@v3
+        with:
+          path: ~/.hgexternalcache
+          key: ${{ runner.os }}-${{ hashFiles('*/external/binaries-list', '*/*/external/binaries-list') }}
+          restore-keys: ${{ runner.os }}-
+
+      - name: Create ${{ matrix.config }}-src zip
+        run: ant $OPTS -quiet build-source-config -Dcluster.config=${{ matrix.config }}
+
+      - name: Extract ${{ matrix.config }}-src zip
+        run: |
+          mkdir tmpbuild && cd tmpbuild
+          unzip -qq ../nbbuild/build/${{ matrix.config }}-src*
+
+      - name: Build from ${{ matrix.config }}-src zip
+        run: |
+          cd tmpbuild
+          ant $OPTS -quiet build -Dcluster.config=${{ matrix.config }}
+
+      # extra round for VSCode which is built with 'release' config
+      - name: Extract ${{ matrix.config }}-src zip for VSCodeExt
+        if: ${{ (matrix.config == 'release') && success() }}
+        run: |
+          rm -Rf tmpbuild && mkdir tmpbuild && cd tmpbuild
+          unzip -qq ../nbbuild/build/${{ matrix.config }}-src*
```

Comment:
The VSCode plugin needs `ant build` to run first - this step looks wrong and I assume will fail - see https://github.com/apache/netbeans/blob/master/java/java.lsp.server/vscode/BUILD.md

Also you'll probably need to set up node.

Reply:
i suppose node is already there otherwise the VSCode job would be failing by now. Updated the workflow to not nuke the tmpbuild folder.

Reply:
:eyes: https://github.com/apache/netbeans/blob/master/.github/workflows/main.yml#L2593 :smile:

Reply:
ok maybe not :), but it would surprise me if node wasn't on the runner yet. Will add the setup line too, thanks!